config/config_test.go

```diff
@@ -52,3 +52,182 @@ func TestTLSConfiguration(t *testing.T) {
 	cfg.RPC.TLSKeyFile = "/abs/path/to/file.key"
 	assert.Equal("/abs/path/to/file.key", cfg.RPC.KeyFile())
 }
+
+func TestBaseConfigValidateBasic(t *testing.T) {
+	cfg := TestBaseConfig()
+	assert.NoError(t, cfg.ValidateBasic())
+
+	// tamper with log format
+	cfg.LogFormat = "invalid"
+	assert.Error(t, cfg.ValidateBasic())
+}
+
+func TestRPCConfigValidateBasic(t *testing.T) {
+	testCases := []struct {
+		testName                     string
+		cfgGRPCMaxOpenConnections    int
+		cfgMaxOpenConnections        int
+		cfgMaxSubscriptionClients    int
+		cfgMaxSubscriptionsPerClient int
+		cfgTimeoutBroadcastTxCommit  time.Duration
+		cfgMaxBodyBytes              int64
+		cfgMaxHeaderBytes            int
+		expectErr                    bool
+	}{
+		{"Valid RPC Config", 1, 1, 1, 1, 1, 1, 1, false},
```

Comment:
From the support perspective, this is not a very good choice. I will try to explain:

Suppose new parameter gets added. We'll have to go through all of the tests and add a new column plus add a new row (/ multiple rows if we want to test multiple scenarious)

Wouldn't it be better to test individual properties like

```go
cfg := DefaultXXXCfg()
cfg.MaxHeaderBytes = -1
assert.Error(t, cfg.ValidateBasic())
```

The current style makes sense when you want to test a combination of properties, but does poor job if you want to test individual properties.

Reply:
It wasn't clear to me which structs would be in common flux and which would be somewhat evergreen. Can you specify which ones you'd prefer to have tested this way?

Reply:
Can you please provide the additional requested information so that this can move forward?

Reply:
> Can you specify which ones you'd prefer to have tested this way?

All the config structs: RPCConfig, P2PConfig, MempoolConfig, ConsensusConfig, ...